UPGRADING.md

```diff
@@ -1,316 +1,277 @@
-# Upgrading
+Upgrading Guide
+===============
 
-Guide to ease migrations between significant changes
+Upgrade from 5.X to 6.X
+-----------------------
 
-## v5 -> v6
+__This version contains many breaking changes__. It is part of an effort to unify our notifier libraries across platforms, making the user interface more consistent, and implementations better on multi-layered environments where multiple Bugsnag libraries need to work together (such as React Native).
 
-### `BugsnagConfiguration` class
+### Key points
 
-#### Instantiation
+- Several configuration options have been renamed and can now load configuration options from a `.plist`.
+- Callback blocks have been expanded to breadcrumbs and sessions.
 
-Initializing a configuration requires a valid API key.
+More details of these changes can be found below and full documentation is available online
+([iOS](https://docs.bugsnag.com/platforms/ios) |
+ [macOS](https://docs.bugsnag.com/platforms/macos) |
+ [tvOS](https://docs.bugsnag.com/platforms/tvos)).
+
+### Bugsnag Client
+
+#### Starting Bugsnag
+
+You can now start Bugsnag using your application's `plist` file with configuration values, including your API key. The simplest start-up code is therefore:
 
 ```objc
-NSError *error;
-BugsnagConfiguration *config = [[BugsnagConfiguration alloc] initWithApiKey:@"YOUR API KEY HERE"];
+[Bugsnag start];
```

Comment:
IMO it'd be ok to just pick one language for examples - iOS developers have to read both languages so they'll be able to understand Swift, and it simplifies the flow.

Reply:
@duncanhewett - do you have thoughts on whether we use just Obj-C or Swift? For the most part the changes are equivalent, but I don't think it loses much in readability to have both.

Reply:
I'm happy either way but it doesn't seem too distracting to have both.